crypto/s2n_stream_cipher_rc4.c

```diff
@@ -21,20 +21,30 @@
 #include "utils/s2n_blob.h"
 #include "utils/s2n_safety.h"
 
-static uint8_t s2n_stream_cipher_rc4_available()
+static const EVP_CIPHER *s2n_evp_rc4()
 {
 #ifdef S2N_LIBCRYPTO_SUPPORTS_EVP_RC4
+    return EVP_rc4();
+#else
+    return NULL;
+#endif
+}
+
+static uint8_t s2n_stream_cipher_rc4_available()
+{
     if (s2n_is_in_fips_mode()) {
         return 0;
-    } else {
-        return (EVP_rc4() ? 1 : 0);
     }
-#else
-    return 0;
-#endif /* S2N_LIBCRYPTO_SUPPORTS_EVP_RC4 */
+    /* RC4 MIGHT be available in Openssl-3.0, depending on whether or not the
+     * "legacy" provider is loaded. However, for simplicity, assume that RC4
+     * is unavailable.
+     */
+    if (S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)) {
+        return 0;
+    }
+    return (s2n_evp_rc4() ? 1 : 0);
```

Comment:
should this fn return take a `*bool` arg and modify it? We are using 1 as true and 0 as false and possibly hiding errors.

Reply:
I'd love to, but the definition of this method is set by [`struct s2n_cipher`](https://github.com/aws/s2n-tls/blob/main/crypto/s2n_cipher.h#L85). Changing that would involve updating all the ciphers and is definitely out of scope of this PR.

Reply:
https://github.com/aws/s2n-tls/issues/3983